bxt-strafe/src/steps.rs

```diff
@@ -410,7 +410,21 @@ impl<S: Step> Step for Strafe<S> {
                     _ => 0.,
                 };
 
-                let vel_yaw = state.player.vel.y.atan2(state.player.vel.x);
+                let mut vel_yaw = state.player.vel.y.atan2(state.player.vel.x);
+
+                // Constant yawspeed.
+                // TODO: if yawspeed is too high, prediction will be wrong.
+                if let StrafeType::ConstYawspeed(yawspeed) = type_ {
+                    // Change vel_yaw in-place for convenience.
```

Comment:
So I'm looking at this code and thinking how to restructure it in a way that makes sense. And it got me thinking, what input is even expected with constant yawspeed strafing? Like, sure, the camera yaw should turn at a constant rate, but which input vector is expected? As an example, with vectorial strafing we can pick any other strafe type (max angle, max accel), then add a `change target_yaw to 180 over 1 s` and technically have constant camera yaw turn rate while the input will be anything else. This is certainly not what we expect though.

I guess the expected result here is always inputting 90 degrees to the left when turning left and 90 degrees to the right when turning right, equivalent to holding A and D button respectively.

Right now the code seems to set the velocity yaw as turned by the yaw delta, then apply a max accel theta to it. When testing with high aa especially, this will most of the time result in a 90 degree input vector indeed, however at lower speeds or aa, when theta is not 90 degrees, it will result in an input vector for which I don't see any reasonable meaning or explanation, so it will probably do something unexpected.

Therefore, I think for constant yawspeed, theta should be hardcoded to PI/2 (90 degrees).

As for a better way to structure this code, you could add an if branch across this part that computes and deals with vel_yaw and theta, and in the constant yawspeed part of the branch, instead of even using vel_yaw or theta, just `let camera_yaw = angle_mod_rad(input.yaw +- max_yaw_delta)` then set forward = 0 and side = +-constant whatever Vct sets it to.

Reply:
Might need a similar change in hlstrafe to avoid using theta for constyawspeed for similar reasoning.